### DISCUSSION TITLE
restarting an old quasar vue project

### DISCUSSION THREAD

**Discussion by abid008** [quasarframework/quasar#10307] [Quasar App Extensions]
Hello all,

I have never worked on a vue or quasar project, and the project I am starting to work on is a project that was done few years back, the team that worked on this project, has long gone. Anyway I checked out the code and per read-me instruction, trying to invoke 

quasar build

and getting following:

quasar build

 Error Unknown command "build"

  ___
 / _ \ _   _  __ _ ___  __ _ _ __
| | | | | | |/ _` / __|/ _` | '__|
| |_| | |_| | (_| \__ \ (_| | |
 \__\_\\__,_|\__,_|___/\__,_|_|

  Running @quasar/cli v1.2.1

  Example usage
    $ quasar <command> <options>

  Help for a command
    $ quasar <command> --help
    $ quasar <command> -h

  Options
    --version, -v Print Quasar CLI version

  Commands
    create    Create a project folder
    info      Display info about your machine
                   (and your App if in a project folder)
    upgrade   Check (and optionally) upgrade Quasar packages
                   from a Quasar project folder
    serve     Create an ad-hoc server on App's distributables
    help, -h  Displays this message

  --------------
  => IMPORTANT !
  => Trigger this inside of a Quasar project (and npm/yarn install) for more commands.

any faster help will be appreciated

**Comment by MilosPaunovic** [comment]
Run `quasar info` command and paste output here so there is a little bit more information.

**Reply by abid008** [reply]
quasar info

Operating System        Windows_NT(10.0.17763) - win32/x64
NodeJs                  14.17.3

Global packages
  NPM                   6.14.13
  yarn                  1.22.11
  @quasar/cli           1.2.1
  @quasar/icongenie     Not installed
  cordova               Not installed

Networking
  Host                  925702DC
  Ethernet 7            172.25.8.209

**Reply by MilosPaunovic** [reply]
Have you ran `npm install`?

**Reply by abid008** [reply]
and here is the package.json

{
  "name": "bits-ui",
  "productName": "BITS UI",
  "version": "0.0.1",
  "private": true,
  "description": "Elections Ballot Information Tracking System",
  "author": "Your Name <your@email.com>",
  "scripts": {
    "clean": "node build/script.clean.js",
    "dev": "node build/script.dev.js",
    "build": "node build/script.build.js",
    "lint": "eslint --ext .js,.vue src"
  },
  "dependencies": {
    "@quasar/cli": "^1.2.1",
    "axios": "^0.17.0",
    "babel-runtime": "^6.25.0",
    "crypto-js": "^3.1.9-1",
    "js-cookie": "^2.2.0",
    "moment": "^2.19.1",
    "quasar-extras": "0.x",
    "quasar-framework": "^0.14.4",
    "vue": "^2.5.0",
    "vue-moment": "^2.1.0",
    "vue-router": "^3.0.1",
    "vuelidate": "^0.6.1",
    "vuex": "^3.0.0"
  },
  "devDependencies": {
    "autoprefixer": "^6.4.0",
    "babel-core": "^6.0.0",
    "babel-eslint": "^7.0.0",
    "babel-loader": "^7.1.2",
    "babel-plugin-transform-runtime": "^6.0.0",
    "babel-preset-es2015": "^6.0.0",
    "babel-preset-stage-2": "^6.0.0",
    "colors": "^1.1.2",
    "connect-history-api-fallback": "^1.1.0",
    "css-loader": "^0.28.7",
    "es6-promise": "^4.1.1",
    "eslint": "^4.8.0",
    "eslint-config-standard": "^10.2.1",
    "eslint-friendly-formatter": "^3.0.0",
    "eslint-loader": "^1.9.0",
    "eslint-plugin-html": "^3.2.2",
    "eslint-plugin-import": "^2.7.0",
    "eslint-plugin-node": "^5.2.0",
    "eslint-plugin-promise": "^3.5.0",
    "eslint-plugin-standard": "^3.0.1",
    "eventsource-polyfill": "^0.9.6",
    "express": "^4.16.1",
    "extract-text-webpack-plugin": "^3.0.0",
    "file-loader": "^0.11.1",
    "friendly-errors-webpack-plugin": "^1.1.3",
    "glob": "^7.1.2",
    "html-webpack-plugin": "^2.30.1",
    "http-proxy-middleware": "^0.17.0",
    "json-loader": "^0.5.7",
    "opn": "^5.0.0",
    "optimize-css-assets-webpack-plugin": "^3.2.0",
    "postcss-loader": "^1.0.0",
    "progress-bar-webpack-plugin": "^1.10.0",
    "purify-css": "^1.2.6",
    "shelljs": "^0.7.0",
    "stylus": "^0.54.5",
    "stylus-loader": "^3.0.1",
    "url-loader": "^0.5.7",
    "vue-loader": "^13.0.5",
    "vue-style-loader": "^3.0.3",
    "vue-template-compiler": "^2.5.0",
    "vuex-persistedstate": "^2.4.2",
    "webpack": "^3.6.0",
    "webpack-dev-middleware": "^1.12.0",
    "webpack-hot-middleware": "^2.19.1",
    "webpack-merge": "^4.1.0"
  }
}

**Reply by abid008** [reply]
yes I have done "npm install" several times, also have tried removed node-modules and lock files

**Reply by abid008** [reply]
> Have you ran `npm install`?

yes I have run "npm install" and also have tried removing node-modules folder and lock files

**Reply by MilosPaunovic** [reply]
Have you tried running any of the scripts defined in `package.json`, e.g. `npm run dev`?

**Reply by abid008** [reply]
> Have you tried running any of the scripts defined in `package.json`, e.g. `npm run dev`?

I have tried:

node build/script.dev.js

and this failed

**Reply by abid008** [reply]
let me try "npm run dev"

**Comment by abid008** [comment]
npm run dev is giving a lot of warnings:

(node:16848) Warning: Accessing non-existent property 'cd' of module exports inside circular dependency
(node:16848) Warning: Accessing non-existent property 'chmod' of module exports inside circular dependency
(node:16848) Warning: Accessing non-existent property 'cp' of module exports inside circular dependency
(node:16848) Warning: Accessing non-existent property 'dirs' of module exports inside circular dependency
(node:16848) Warning: Accessing non-existent property 'pushd' of module exports inside circular dependency

**Reply by MilosPaunovic** [reply]
My best guess is that this is an issue regarding versions of `node`, `npm` or some other dependency, as Quasar `0.14.4` is really old.

You should find official docd for that version and check requirements.

**Reply by abid008** [reply]
the "npm run dev" is not working, but "npm run build" compiled the app successfully, it says that I should use an http server, which one should I use? like tomcat or nginx?

**Reply by MilosPaunovic** [reply]
That is up to you.

**Reply by abid008** [reply]
so you mean both will work, right? I already have tomcat on my machine I will try that. Thanks for the help.

**Reply by MilosPaunovic** [reply]
As long as you set it up the way your app needs, there is no reason why any server won't work.

**Reply by abid008** [reply]
any idea, where would I specify the API host and port?

**Reply by abid008** [reply]
I found the answer thanks!

**Comment by ontwikkelfabriek** [comment]
@abid008 

Could you tell us if a Quasar **v0.14** app is still usable/working as of 2021?

**Reply by abid008** [reply]
yes, it still working!

**Reply by abid008** [reply]
as you said above this app was written a long time ago, and did not require any update or bug fix  until now.

**Reply by abid008** [reply]
and today I am using 

node 14.17
quasar cli 1.22.11

**Reply by glittle** [reply]
I'm in a similar predicament.  I have a working system that was built with cli 0.6.5. My system is updated to use 1.0.5 and now I can't build my old project.

**Reply by ontwikkelfabriek** [reply]
@glittle 

> I have a working system that was built with cli 0.6.5. My system is updated to use 1.0.5 and now I can't build my old project.

So what is preventing you from installing cli version 0.6.5?


With this command you can list all available versions of quasar-cli (it appears 0.6.5 is still available):
```
npm view quasar-cli versions
```

Uninstall the old cli version
```
npm uninstall -g quasar-cli
```

Install the 0.6.5 version:
```
npm install -g @quasar/cli@0.6.5
```

**Reply by glittle** [reply]
Thank you. Someone in the discord channel helped me find the 0.6.5 version and I've been able to do this. The trick was to use "quasar-cli", not "@quasar/clr"
